dive_core/available_gpu_time.h

```diff
@@ -0,0 +1,101 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#pragma once
+
+#include <filesystem>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace Dive
+{
+
+/*
+AvailableGpuTiming parses CSV format file (gpu_time.csv) produced by looping GFXR replay into
+available timing info statistics.
+
+TODO: Retrieve GPU timing info from device to host through network/ rather than using a CSV file.
+*/
+
+namespace
+{
+constexpr char kExpectedHeader[] = "Type,Id,Mean [ms],Median [ms]";
+constexpr int  kExpectedColumns = 4;
+}  // namespace
+
+class AvailableGpuTiming
+{
+public:
+    enum class ObjectType : uint8_t
+    {
+        kFrame = 0,
+        kCommandBuffer = 1,
+        kRenderPass = 2,
+        nObjectTypes = 3,
+    };
+
+    struct Entry
+    {
+        ObjectType object_type;
+        uint32_t   per_frame_id;
+    };
+
+    struct Stats
+    {
+        float mean_ms;
+        float median_ms;
+    };
+
+    AvailableGpuTiming() = default;
+    ~AvailableGpuTiming() = default;
+
+    // Load statistics from a CSV file and flag as loaded afterwards
+    bool LoadFromCsv(const std::filesystem::path& file_path);
+
+    // Load statistics from a string and flag as loaded afterwards
+    // For unit testing
+    bool LoadFromString(const std::string& full_text);
+
+    // Get the statistic info with the ObjectType and the id (nth object of type ObjectType)
+    // If object_type is kFrame, the object_id value is disregarded
+    std::optional<Stats> GetStats(ObjectType object_type, uint32_t object_id) const;
```

Comment:
i find overloading GetStats is a bit confusing here: 
In GetStats(ObjectType object_type, uint32_t object_id), it means the 0-based
In GetStats(uint32_t object_id), it means the 1-based row index from the original file

Comment helps here, but might be better to have different names, something like
```
std::optional<Stats> GetStatsByRow(uint32_t one_based_row_id) const;
std::optional<Stats> GetStatsByType(ObjectType object_type, uint32_t per_type_id) const;
```